hilti/runtime/src/types/bytes.cc

```diff
@@ -171,14 +171,14 @@ integer::safe<uint64_t> Bytes::toUInt(uint64_t base) const {
 }
 
 int64_t Bytes::toInt(ByteOrder byte_order) const {
-    auto i = toUInt(byte_order);
+    auto i = toUInt(byte_order); // throws on size == 0 or size > 8
     auto size_ = static_cast<uint64_t>(size());
 
-    if ( i & (1U << (size_ * 8 - 1)) ) {
+    if ( i & (1ULL << (size_ * 8 - 1)) ) {
```

Comment:
This should work now, but I still find the mixing of types with explicit (`uint64_t` and minimum sizes (`unsigned long long`) confusing. WDYT about about using sized types throughout, at least were we care, e.g.,

```suggestion
    if ( i & (UINT64_C(1) << (size_ * 8 - 1)) ) { // requires `#include <cstdint>`
```